src/lib.rs

```diff
@@ -0,0 +1,83 @@
+use laz::las::file::{read_vlrs_and_get_laszip_vlr, QuickHeader};
+use laz::las::laszip::LasZipDecompressor;
+use wasm_bindgen::prelude::*;
+use std::io::Seek;
+extern crate console_error_panic_hook;
+
+#[wasm_bindgen]
+pub struct WasmQuickHeader {
+    pub major: u8,
+    pub minor: u8,
+    pub offset_to_points: u32,
+    pub num_vlrs: u32,
+    pub point_format_id: u8,
+    pub point_size: u16,
+    pub num_points: u64,
+    pub header_size: u16,
+}
+
+impl From<QuickHeader> for WasmQuickHeader {
+    fn from(other: QuickHeader) -> WasmQuickHeader {
+        WasmQuickHeader {
+            major: other.major,
+            minor: other.minor,
+            offset_to_points: other.offset_to_points,
+            num_vlrs: other.num_vlrs,
+            point_format_id: other.point_format_id,
+            point_size: other.point_size,
+            num_points: other.num_points,
+            header_size: other.header_size,
+        }
+    }
+}
+
+#[wasm_bindgen]
+pub fn get_header(buf: js_sys::Uint8Array)  -> std::result::Result<WasmQuickHeader, JsValue> {
+    // initialize debugging
+    console_error_panic_hook::set_once();
+
+    // copy header bytes into wasm memory
+    let mut body = vec![0; buf.length() as usize];
+    buf.copy_to(&mut body[..]);
+    // cursor to wrap the bytes
+    let mut cursor = std::io::Cursor::new(body);
+    let hdr = QuickHeader::read_from(&mut cursor).unwrap();
+    Ok(WasmQuickHeader::from(hdr))
+}
+
+#[wasm_bindgen]
+pub struct WasmLasZipDecompressor {
+    decompressor: LasZipDecompressor<'static, std::io::Cursor<Vec<u8>>>,
+}
+
+impl WasmLasZipDecompressor {
+    pub fn new(source: Vec<u8>) -> Self {        
+        let mut cursor = std::io::Cursor::new(source);
+
+        let hdr = QuickHeader::read_from(&mut cursor).unwrap();
+        
+        cursor.seek(std::io::SeekFrom::Start(hdr.header_size as u64));
```

Comment:
`seek` result is not used, we should at least use `unwrap()` until we properly create a Result/Exception